### PR TITLE
Raw & Serialized SQLiteStorage

### LINKED ISSUE #3423: SQLite serialization should be done through a proxy
## Problem Definition

https://github.com/raiden-network/raiden/blob/d50045c7aec544d225dcddc4fb2ee81f35e1949e/raiden/utils/migrations/v16_to_v17.py#L5-L9

The snippet above is that a migration needed to get the list of snapshots in the database. Another migration i recently added in PR (#2832) required the same thing so i had to duplicate some code. One would argue that we can place these functions inside the storage (SQliteStorage). However, the migrations don't want to operate on the "deserialized" version of the snapshots but rather the raw json string. This is because if the definition for the state / state change / event changes, the raw JSON stored into the older version databases will not deserialize properly.
The need for a manually fetching the snapshots from the database not through the storage was because the storage uses the serializer by default.

### Proposed solution

1. Decouple serialization/deserialization from storage where the storage only returns / gets raw data.
2. Create a proxy class from which access to the storage can be done with the added serialization / deserialization capabilities.

Related issue #3405

### PR DESCRIPTION
**Pull request by rakanalh**
## What changed?
- Separate serialization logic into `SerializedSQLiteStorage` while `SQLiteStorage` returns raw data.
- Introduced `transaction` method in storage to start an explicit transaction. 
- Introduced `maybe_commit` will only commit if we're outside an explicit transaction.
- Introduced `get_snapshot` and `update_snapshot` in  `SQLiteStorage`

Resolves #3423

### CODE REVIEW

**Comment by jomuel** [comment]
Reviewing this now

**Comment by rakanalh** [comment]
Thanks @jomuel

File: raiden/utils/upgrades.py [hunk 1]
```diff
@@ -119,27 +106,27 @@ def run(self):
...
                         )
 
-                    update_version(cursor, RAIDEN_DB_VERSION)
+                    update_version(storage, RAIDEN_DB_VERSION)
                     # Prevent the upgrade from happening on next restart
                     self._backup_old_db(old_db_filename)
-                except RaidenDBUpgradeError:
-                    self._delete_current_db()
-                    raise
+            except Exception as e:
```
Comment (LefterisJP) [inline raiden/utils/upgrades.py]: @rakanalh Why did you remove the specific `RaidenDBUpgradeError` exception catching from here and put a catch-all exception? Catch-alls are something to be avoided.

File: tools/debugging/migrate_db.py [hunk 2]
```diff
@@ -25,7 +24,7 @@ def upgrade_db(current_version: int, new_version: int):
     )
     try:
         manager.run()
-    except (RaidenDBUpgradeError, InvalidDBData) as e:
+    except Exception as e:
```
Comment (LefterisJP) [inline tools/debugging/migrate_db.py]: same here

File: raiden/utils/upgrades.py (hunk 1 above)
Comment (rakanalh) [reply raiden/utils/upgrades.py]: In the case of migrations, we agreed that we wanted to catch all exceptions so that proper rollback handling is done in case of ANY failure.

Comment (rakanalh) [reply raiden/utils/upgrades.py]: The class at that point didn't really make much more sense so i removed it
